framework/src/st_things/things_stack/framework/things_req_handler.c

```diff
@@ -190,6 +190,96 @@ OCEntityHandlerResult send_response(OCRequestHandle request_handle, OCResourceHa
 	return eh_result;
 }
 
+static OCEntityHandlerResult convert_ap_infor(things_resource_s *target_resource, access_point_info_s *p_list, int list_cnt)
+{
+	things_representation_s *rep = NULL;
+
+	if (list_cnt == 0 || p_list == NULL) {
+		THINGS_LOG_E(TAG, "Scaned AP list is NULL ");
+		return OC_EH_ERROR;
+	}
+
+	if (target_resource->rep == NULL) {
+		rep = things_create_representation_inst(NULL);
+		target_resource->things_set_representation(target_resource, rep);
+	} else {
+		rep = target_resource->rep;
+	}
+	things_representation_s **child_rep = (things_representation_s **)things_malloc(list_cnt * sizeof(things_representation_s *));
+	access_point_info_s *wifi_scan_iter = p_list;
+	int iter;
+	for (iter = 0; iter < list_cnt; ++iter) {
+		THINGS_LOG_V(TAG, "e_ssid : %s", wifi_scan_iter->e_ssid);
+		// THINGS_LOG_V(TAG, "security_key : %s", wifi_scan_iter->security_key);
+		// THINGS_LOG_V(TAG, "auth_type : %s", wifi_scan_iter->auth_type);
+		// THINGS_LOG_V(TAG, "enc_type : %s", wifi_scan_iter->enc_type);
+		// THINGS_LOG_V(TAG, "channel : %s", wifi_scan_iter->channel);
+		THINGS_LOG_V(TAG, "signal_level : %s", wifi_scan_iter->signal_level);
+		THINGS_LOG_V(TAG, "bss_id : %s", wifi_scan_iter->bss_id);
+		child_rep[iter] = things_create_representation_inst(NULL);
+		// child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_CHANNAL, wifi_scan_iter->channel);
+		// child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_ENCTYPE, wifi_scan_iter->enc_type);
+		child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_MACADDR, wifi_scan_iter->bss_id);
+		child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_MAXRATE, "MAX_RATE");
+		child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_RSSI, wifi_scan_iter->signal_level);
+		// child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_SECTYPE, wifi_scan_iter->auth_type);
+		child_rep[iter]->things_set_value(child_rep[iter], SEC_ATTRIBUTE_AP_SSID, wifi_scan_iter->e_ssid);
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	rep->things_set_arrayvalue(rep, SEC_ATTRIBUTE_AP_ITEMS, list_cnt, child_rep);
+	for (iter = 0; iter < list_cnt; iter++) {
+		/*! Changed by st_things for memory Leak fix
+		 */
+		things_release_representation_inst(child_rep[iter]);
+		child_rep[iter] = NULL;
+	}
+	things_free(child_rep);
+	child_rep = NULL;
+	return OC_EH_OK;
+}
+
+/**
+ * This functions will be deprecated~!!!
+ */
+static OCEntityHandlerResult get_ap_scan_result(things_resource_s *target_resource)
+{
+	THINGS_LOG_D(THINGS_DEBUG, TAG, THINGS_FUNC_ENTRY);
```

Comment:
Would you please check first argument.

Reply:
Can you specify what is the thing I should check in first argument?

Reply:
@harish-191 Could you please remove first argument?
THINGS_LOG_D requires first argument is TAG.

Reply:
ok